tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensors.java

```diff
@@ -30,12 +30,20 @@ public static String toString(Tensor tensor) {
   }
 
   /**
+   * Returns a String representation of a tensor's data. If the output is wider than {@code
+   * maxWidth} characters, it is truncated and "{@code ...}" is inserted in place of the removed
+   * data.
+   *
    * @param tensor   a tensor
    * @param maxWidth the maximum width of the output in characters ({@code null} if unlimited). This
    *                 limit may surpassed if the first or last element are too long.
    * @return the String representation of the tensor
    */
   public static String toString(Tensor tensor, Integer maxWidth) {
+    if (tensor instanceof RawTensor) {
+      System.out.println("Got rawTensor: " + tensor);
+      tensor = ((RawTensor) tensor).asTypedTensor();
+    }
     if (!(tensor instanceof NdArray)) {
       throw new AssertionError("Expected tensor to extend NdArray");
```

Comment:
Can you make this message more descriptive?  Mention the tensor's dtype, the typed class (i.e. `tensor.getClass()`), and that it isn't mapped to an ndarray and thus can't be printed.  Maybe mention that you can use `tf.print`, too.

Reply:
I will add more debugging information but keep in mind that this error is not meant to be user facing (hence `AssertionError` versus a normal exception). If we see this error it is because someone added a new type of tensor and forgot to update `Tensors` to support it. This corresponds to an internal error on our part. At least, that was my intent.